.circleci/config.yml

```diff
@@ -536,6 +577,91 @@ jobs:
     steps:
       - checkout
 
+  importers:
+    parameters:
+      python_version_major:
+        type: integer
+        default: 3
+      python_version_minor:
+        type: integer
+        default: 8
+      shard:
+        type: string
+        default: "core"
+      toxenv:
+        type: string
+      codecov_flags:
+        type: string
+        default: ""
+      notify_on_failure:
+        type: boolean
+        default: false
+      xdist:
+        type: integer
+        default: 3
+      tox_args:
+        type: string
+        default: ""
+    machine: true
+    resource_class: xlarge
+    working_directory: ~/project
+    environment:
+      SHARD: << parameters.shard >>
+    steps:
+      - checkout
+      - run:
+          name: Install system deps
+          command: |
+            sudo apt-get update && sudo apt-get install -y libsndfile1 ffmpeg
+      - gcloud/install
+      - setup_gcloud
+      - run:
+          name: Decode GCP Service Account JSON
+          command: echo $GCP_SERVICE_ACCOUNT_JSON_ENCODED | base64 --decode > ${HOME}/gcp-key.json
+      - run:
+          name: Authenticate with GCP Artifact Registry
+          command: |
+            echo $GCP_SERVICE_ACCOUNT_JSON_DECODED | docker login -u _json_key --password-stdin https://<< pipeline.parameters.gcp_project >>-docker.pkg.dev
```

Comment:
Instead of this custom code, you probably need to do something like this:

https://github.com/wandb/wandb/blob/40b5826a97562514ee3a006382fd7076e7db4efa/.circleci/config.yml#L566-L569

Is this what you're looking for? https://circleci.com/docs/using-docker/#using-multiple-docker-images

Reply:
Currently, the `importers` job uses a `machine` executor because it needs to spin up two test-containers and the second test-container needs a port remapping (otherwise the two container ports will collide).  AFAIK there's no way to do the port remapping otherwise in CircleCI.

Reply:
It looks like port remapping isn't available, but you can set the hostname using the `name` parameter: https://circleci.com/docs/configuration-reference/#docker. Something like this:

```yaml
docker:
  - image: ...
  - image: ...
    name: localhost-wandb-2
```

Then when creating the `WandbServerSettings` in your test, you should be able to set the `url` to `http://localhost-wandb-2`.

Reply:
Thanks, I think that works!  How about something like this?
https://github.com/wandb/wandb/pull/6897/files/e40954eacdfcb94bd48fc1e09e04858c3547b3aa..1637a99455e4677c97925962512ceca194e925a5